### DISCUSSION TITLE
The Truth About DreamBerd

### DISCUSSION THREAD

**Discussion by TodePond** [TodePond/GulfOfMexico#460] [Announcements]
[Last week](https://github.com/TodePond/DreamBerd/releases/tag/v0.0), we shut down the company, so there's no need to stay in character anymore.

For the first time ever, these release notes will contain no jokes, no sarcasm, and no humour of any kind. Instead, it's time to answer all the questions you ask us most. Questions like...

• Is DreamBerd a joke?
• What is DreamBerd making fun of?
• Will you accept my PR full of offensive jokes?

It's time to answer them all. No more lies! Here's the truth:

# The Truth About DreamBerd

## Is DreamBerd a joke?

Yes.

## What is DreamBerd making fun of?

DreamBerd is making fun of **JavaScript**.

JavaScript is a terrible language, full of terrible design decisions. Just look at this abomination:

```js
!!!!!!!true
```

Yeah... You read that right. JavaScript lets you put multiple exclamation marks at the start of a line. Dreamberd mocks this by allowing you to put multiple exclamation marks at the end of the line.

*That's the kind of high-level satire you can expect from DreamBerd.*

Here's another one:

```js
typeof NaN === "number"
```

Yeah... You read that right. JavaScript thinks that 'not a number' is a 'number'. You probably didn't know this.

*Not everyone will get these jokes. You have to be thinking quite laterally.*

## Will you accept my PR full of offensive jokes?

No.

---

Tune in next week for more DreamBerd release notes.

> If you love anti-javascript humour, then you'll *love* my video about how bad it is. Watch (or re-watch) it [here](https://youtu.be/ZMklf0vUl18).

<hr /><em>This discussion was created from the release <a href='https://github.com/TodePond/DreamBerd/releases/tag/vNaN.NaN'>The Truth About DreamBerd</a>.</em>

**Comment by radeeyate** [comment]
glad to know dreamberd is back!! 🚀 👀

**Comment by kerbalis** [comment]
Happy

**Reply by l11111111111111** [reply]
Happy

**Reply by marvinborner** [reply]
Very happy

**Reply by radeeyate** [reply]
rejoiced

**Reply by atomicapple0** [reply]
https://www.youtube.com/watch?v=KN3IemyqJJw

**Comment by mybearworld** [comment]
> Yeah... You read that right. JavaScript lets you put multiple exclamation marks at the start of a line. Dreamberd mocks this by allowing you to put multiple exclamation marks at the end of the line.

What about `;;;;;;;true`? This is valid syntax in DreamBerd and returns `false`.

<sup>(Side note: This returns `true` in JavaScript.)</sup>

**Reply by iqnite** [reply]
JavaScript was made to make fun of DreamBerd. Its creator once said
> Yeah... You read that right. DreamBerd lets you put multiple semicolons at the start of a line. JavaScript mocks this by allowing you to put multiple semicolons at the end of the line.

in an interview with c't.

**Comment by darrowv** [comment]
What's wrong with NaN being considered as a number type if it really is according to the IEEE 754 standard? EcmaScript employs this standard, so wouldn't it be worse if JavaScript were inconsistent in this particular part? Just interesting

**Reply by TodePond** [reply]
Yeah I think you're onto something here

**Reply by MaddyGuthridge** [reply]
Don't think you're thinking about it literally enough tbh

**Comment by jozsefsallai** [comment]
you forgot to mention using floats as array indexes (which is implemented better in DreamBerd)

this is perfectly valid JavaScript:
```js
const arr = ['hello', 'world'];
arr[0.5] = '???';
```
in fact you can also use negative numbers if that's what you're into, but I think we all should just stay positive

**Reply by phoopee3** [reply]
How can you be down on JS when it has these types of surprises?! You can double your memory by changing the step count in your arrays.

**Comment by bhardvaja** [comment]
![banana-cat-crying](https://github.com/TodePond/DreamBerd/assets/63110845/38d79a76-3f02-4237-a1d4-a7c85dc11506)

so sad

**Comment by TodePond** [comment]
Wat

haha

**Reply by mybearworld** [reply]
haha

**Reply by iqnite** [reply]
haha

**Reply by skifli** [reply]
haha

**Reply by radeeyate** [reply]
haha

**Comment by skifli** [comment]
Can't forget that an empty array is the same as an empty string `[] == '' // true`.

**Reply by mybearworld** [reply]
See, that's where DreamBerd shines! If you don't want this behaviour, you can just not have it: `[] === ''` And if you do, you can: `[] == ""`


oh wait

**Reply by skifli** [reply]
But I can't be tasked with choosing which method for such a vital decision. Surely the programming language should decide so I can blame any consequences on it?

**Reply by TodePond** [reply]
I think you're on the right path here, keep going.

**Comment by MaddyGuthridge** [comment]
Will you start up the company again if I ask nicely?

**Reply by TodePond** [reply]
Yes

**Reply by MaddyGuthridge** [reply]
Please start the company again 🥺

**Reply by TodePond** [reply]
Ok sure!

**Comment by JustLeif** [comment]
console.log(typeof 'JavaScript isn\'t that bad?' === false)
=> true

**Comment by mybearworld** [comment]
unrelated but did GitHub just change how the writing comments UI looks? interesting

**Reply by spifory** [reply]
They did! See https://github.blog/changelog/2023-10-31-the-comment-box-becomes-more-consistent-and-accessible :)
